test/fail_compilation/fail2789.d

```diff
@@ -1,11 +1,5 @@
 /*
 https://issues.dlang.org/show_bug.cgi?id=18385
-
-TEST_OUTPUT:
----
-fail_compilation/fail2789.d(15): Error: function `fail2789.A2789.m()` conflicts with previous declaration at fail_compilation/fail2789.d(10)
-fail_compilation/fail2789.d(25): Error: function `fail2789.A2789.m()` conflicts with previous declaration at fail_compilation/fail2789.d(10)
----
```

Comment:
I'm confused - how come these functions don't conflict? D doesn't allow overloading on return types alone.

Specifically these two:

```D
class C
{
    int m();
    float m();
}
```
overloading on `const` is ok and the `static` `m` is also fine.

Reply:
The current implementation is tied to the generated mangling which might include the return type. (AFAIC this was done because functions with identical manglings cause glue/backend errors for ldc/gdc).

Guess this should eventually be reworked to compare the `FuncDeclaration`s s.t. it can detect these invalid overloads.

Reply:
Fixed with the new implementation